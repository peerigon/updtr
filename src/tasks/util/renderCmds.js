export default function renderCmds(updtr, updateTasks) {
    const cmds = updtr.cmds;
    const registry = updtr.config.registry;

    return {
        update: cmds.install({
            registry,
            modules: updateTasks.map(updateTask => ({
                name: updateTask.name,
                version: updateTask.updateTo,
            })),
        }),
        test: cmds.test(),
        rollback: cmds.install({
            registry,
            modules: updateTasks.map(updateTask => ({
                name: updateTask.name,
                version: updateTask.rollbackTo,
            })),
        }),
    };
}
