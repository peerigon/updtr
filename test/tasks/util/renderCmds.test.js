import renderCmds from "../../../src/tasks/util/renderCmds";
import FakeUpdtr from "../../helpers/FakeUpdtr";

describe("renderCmds", () => {
    test("should return an object with pre-rendered commands", () => {
        const updtr = new FakeUpdtr({
            registry: "http://example.com",
        });
        const updateTasks = [
            {
                name: "module-1",
                updateTo: "2.0.0",
                rollbackTo: "1.0.0",
            },
            {
                name: "module-2",
                updateTo: "2.0.0",
                rollbackTo: "1.0.0",
            },
        ];

        expect(renderCmds(updtr, updateTasks)).toMatchSnapshot();
    });
});
