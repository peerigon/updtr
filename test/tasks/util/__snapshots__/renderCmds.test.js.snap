// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderCmds should return an object with pre-rendered commands 1`] = `
Object {
  "rollback": "npm install --registry http://example.com module-1@1.0.0 module-2@1.0.0",
  "test": "npm test",
  "update": "npm install --registry http://example.com module-1@2.0.0 module-2@2.0.0",
}
`;
